Add git-profile extension

tuki0918 opened the pull request:
## Description

This is a profile management extension for using multiple Git users.

features:

- Profile
You can view and edit your Git commit `user.name` and `user.email`.
- My Profiles
You can register multiple profiles and switch between them.

## Screencast

![git-profile-1](https://github.com/user-attachments/assets/301ca2c3-afbe-4559-b062-df519b4e3638)

![git-profile-2](https://github.com/user-attachments/assets/2ccdd5db-deab-40ef-83a8-0d14f0724e39)

![git-profile-3](https://github.com/user-attachments/assets/25ffe466-ad28-4cc9-bde7-55aa25424f56)


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Shouldn't it return to previous view when you save?

![git-profile 2025-02-26 at 12 22 53](https://github.com/user-attachments/assets/799f5cea-3d40-451f-9982-1b28c4686a65)

- What's the difference between `My Profiles` and `Profile`

- Can you update the README so it explains a bit about the extension?

- Use `style={Action.Style.Destructive}` for Delete actions

- Use `shortcut={Keyboard.Shortcut.Common.Remove}` for Delete actions



I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

pernielsentikaer commented:
@greptileai can you give it a spin 😊

tuki0918 commented:
Thank you for the feedback!
Sorry for the delay in making the corrections.

----

> I have now tested your extension, and I have some feedback ready for you:
> Shouldn't it return to previous view when you save?

[x] Fixed. It now returns to the previous view after saving.

> What's the difference between My Profiles and Profile

Renamed due to unclear command names.

- `Profile` -> `Current User`
- `My Profiles` -> `Manage Profiles`

`Current User` displays information about the current Git user.
`Manage Profiles` is a list of profiles maintained internally.

> Can you update the README so it explains a bit about the extension?

[x] Updated the README and corrected the text overall.

> Use style={Action.Style.Destructive} for Delete actions

[x] Fixed.

> Use shortcut={Keyboard.Shortcut.Common.Remove} for Delete actions

[x] Fixed.

pernielsentikaer commented:
When no profiles is added in this view it would probably be better to show a `List.EmptyView` instead (Remember to add an icon for the action (all actions should have))

![git-profile 2025-03-03 at 11 52 26](https://github.com/user-attachments/assets/49fc9735-7980-418a-9b5d-cc81668b993e)

tuki0918 commented:
I appreciate your feedback.

![git-profile-4](https://github.com/user-attachments/assets/0102c940-a8dc-4bee-ada5-37de81d67d5c)
![git-profile-5](https://github.com/user-attachments/assets/31a17629-418a-4ee2-a082-085d680a93df)

pernielsentikaer commented:
That looks much better, thanks for enhancing the UI 😊

pernielsentikaer commented:
@greptileai do you have more?

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
